proxy/src/main/java/com/velocitypowered/proxy/connection/client/InitialLoginSessionHandler.java

```diff
@@ -209,63 +210,54 @@ public boolean handle(EncryptionResponse packet) {
         url += "&ip=" + urlFormParameterEscaper().escape(playerIp);
       }
 
-      ListenableFuture<Response> hasJoinedResponse = server.getAsyncHttpClient().prepareGet(url)
-          .execute();
-      hasJoinedResponse.addListener(() -> {
-        if (mcConnection.isClosed()) {
-          // The player disconnected after we authenticated them.
-          return;
-        }
+      HttpRequest httpRequest = HttpRequest.newBuilder().uri(URI.create(url)).build();
+      HttpClient.newHttpClient().sendAsync(httpRequest, HttpResponse.BodyHandlers.ofString())
```

Comment:
Instead of recreating each time the HttpClient, wouldn't it be better to store it in a static variable or in the ProxyServer?

Reply:
Yea, probably should